### PR TITLE
Added Highlighter::style_for_stack

### PR DESCRIPTION
**Pull request by cmyr**
I... think this is right? 🤷‍♂️

### CODE REVIEW

File: src/highlighting/highlighter.rs [hunk 1]
```diff
@@ -239,6 +239,19 @@ impl<'a> Highlighter<'a> {
             font_style: None,
         }
     }
+
+    /// Returns the fully resolved style for the given stack.
+    ///
+    /// This operation is convenient but expensive. For reasonable performance,
+    /// the caller should be caching results.
+    pub fn style_for_stack(&self, stack: &ScopeStack) -> Style {
```
Comment (trishume) [inline src/highlighting/highlighter.rs]: This should really take a `&[Scope]` like the other `Highlighter` methods, there's no reason for it to take a `ScopeStack`.

Comment (cmyr) [inline src/highlighting/highlighter.rs]: I made it take a `ScopeStack` so that I could use the `bottom_n` method that is so helpfully defined there.

Comment (trishume) [inline src/highlighting/highlighter.rs]: With a slice you can use the `[0..n]` syntax which is shorter, and is actually all the `bottom_n` method does.

**Comment by cmyr** [comment]
okay, made that change. 😎

**Pull request merged**